### DISCUSSION TITLE
What is the license for micropip?

### DISCUSSION THREAD

**Discussion by pombredanne** [pyodide/pyodide#2203] [Q&A]
I could not find an explicit license for micropip ( and FWIW for several other packages in https://github.com/pyodide/pyodide/tree/main/packages ) ... what would it be?
Thanks!

**Comment by hoodmane** [comment]
All in-tree packages are supposed to be licensed as MPL2. I think everything is MPL2 unless otherwise specified.

**Reply by pombredanne** [reply]
@hoodmane Thanks for the quick reply! So I just did a quick diff against pip and micropip seems to contain code copied from pip which is MIT-licensed and not MPL2 afaik.
I also found this comment in packages/micropip/src/micropip/externals/pip/__init__.py 
`Based on https://github.com/pypa/pip/commit/57be6a77c57ab5d512371b5c48d508a7620c3217`
may be it would be good to bring back some of the missing notices and license for this? And possibly clearly mark what is MPL vs. MIT or other licenses?

**Reply by hoodmane** [reply]
Oh yeah everything from `externals/pip` folder is vendored from pip and should have the same license as pip. I think "Based on" really means "copied without change". I agree it might be good to clarify this, I am not sure what the standard practice for vendored code is.

**Reply by pombredanne** [reply]
> I agree it might be good to clarify this, I am not sure what the standard practice for vendored code is.

My preferred practice is to document in details the origin and license including all notices, restoring them if not present.
I do this for instance when I vendor virtualenv (with its vendored deps): https://github.com/nexB/scancode-toolkit/tree/develop/etc/thirdparty  or for these packages: https://github.com/nexB/thirdparty-packages/tree/main/pypi

**Reply by pombredanne** [reply]
> I think "Based on" really means "copied without change". 

It looks adapted a bit.

**Reply by hoodmane** [reply]
I'm not sure I understand what your recommendations are. Would you be willing to open a PR with the appropriate changes?

**Reply by rth** [reply]
Yes, micropip should probably have a separate license file indicating that it's MPL2 with vendored files that are MIT. 

For the other packages, they retain their original license. Maybe we should relicense the patches to be MIT or BSD2 to be more compatible with the original package licenses, I'm not sure what are the implication of MPL2 patches in an MIT/BSD2 package, but I'm not sure that's something we want.
